README.md

```diff
@@ -4,3 +4,8 @@ TumCalProxy [![Build Status](https://travis-ci.org/kordianbruck/TumCalProxy.svg)
 Proxy for the TUM iCal export to remove clutter from it and optimize the output
 
 You can find more information on the about page: https://cal.bruck.me/
+
+To run locally using docker:
+ - run `docker run --rm -it -v /ABSOLUTE/PATH/TO/PROJECT/:/app composer install` to install dependencies
+ - run `docker run --rm -it -v /ABSOLUTE/PATH/TO/PROJECT/:/app composer /app/vendor/bin/phpunit` to execute tests
+ - run `docker run --rm -p 80:80 -v /ABSOLUTE/PATH/TO/PROJECT/:/app webdevops/php-nginx:alpine` to run the webserver on port 80
```

Comment:
Could you move this to a `Dockerfile` please? That should allow the user to run it with only a single command and makes it more visible what is being done during a build.

Reply:
```
FROM composer AS builder
COPY ./ /app
RUN composer install && ./vendor/bin/phpunit
FROM webdevops/php-nginx:alpine
COPY --from=builder /app /app
``` 
something like this would create a docker image that runs the project. Is that what you mean? With the docker run additions I had another developer in mind who wants to change things and test things quickly. With the docker run commands the dependencies will be downloaded to the local folder and are available from there on. Tests can be executed on demand. With the Dockerfile the whole image needs to be rebuild on every change and all dependencies need to be redownloaded.

Reply:
docker builds are cached, so it shouldn't be slower but actually much faster than running the full run command from above :+1:

Reply:
It can only be cached if the source files didn't change. Otherwise the docker daemon can't know if a RUN-instruction will cause the same layer or not. So in most cases after a COPY-instruction nothing will be cached. With `docker run` you don't have to do anything if you change code. The local folder is mounted directly into the webserver. You change your php and reload the page in the browser. Nothing more. Only if you add additional dependencies, you will have to repeat the first step. And even then only the dependencies missing will be downloaded. Please try this out
1. make sure there's no vendor-folder in your project folder and create the Dockerfile with above lines
2. `docker build -t test .`
3. `docker run --rm -p 80:80 test` now it is running
4. `touch test`
5. now rebuild with the changed code (docker doesn't know that the empy test file is useless): `docker build -t test .`
6. `docker run --rm -p 80:80 test` now you can access the empy file from your browser
and now compare with:
1. `rm test`
2. run the three commands from the readme
3. now run `touch test` again.
4. reload the browser, the empty file this there immediately

There's really only one use case for using the Dockerfile to build the image. And that is if someone who is not a developer wants to run a copy of the project. Then it's easy to build the project and run it. I can only think of one reason why one would want to do that and that is if they don't trust you with their calendar.

Reply:
Cool, you have me convinced. You are right, I didn't look too closely at the above command - with mounting the code into the container you absolutely save time.